lxc/console.go

```diff
@@ -0,0 +1,177 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"syscall"
+
+	"github.com/gorilla/websocket"
+
+	"github.com/lxc/lxd/client"
+	"github.com/lxc/lxd/lxc/config"
+	"github.com/lxc/lxd/shared/api"
+	"github.com/lxc/lxd/shared/gnuflag"
+	"github.com/lxc/lxd/shared/i18n"
+	"github.com/lxc/lxd/shared/logger"
+	"github.com/lxc/lxd/shared/termios"
+)
+
+type consoleCmd struct {
+	tty    int
+	escape string
+}
+
+func (c *consoleCmd) showByDefault() bool {
+	return true
+}
+
+func (c *consoleCmd) usage() string {
+	return i18n.G(
+		`Usage: lxc console [<remote>:]<container> [-t] [-e]
+
+Interact with the container's console devices and log.
+
+The default is to attach to the containers /dev/console. To exit the console
+type Ctrl+<escape> q. The default for <escape> is 'a'.`)
+}
+
+func (c *consoleCmd) flags() {
+	gnuflag.IntVar(&c.tty, "tty", 0, i18n.G("The console device to attach to"))
+	gnuflag.StringVar(&c.escape, "escape", "a", i18n.G("The escape character to use when exiting the console via \"Ctrl+<escape> q\" device to attach to"))
+}
+
+func (c *consoleCmd) sendTermSize(control *websocket.Conn) error {
+	width, height, err := termios.GetSize(int(syscall.Stdout))
+	if err != nil {
+		return err
+	}
+
+	logger.Debugf("Window size is now: %dx%d", width, height)
+
+	w, err := control.NextWriter(websocket.TextMessage)
+	if err != nil {
+		return err
+	}
+
+	msg := api.ContainerExecControl{}
+	msg.Command = "window-resize"
+	msg.Args = make(map[string]string)
+	msg.Args["width"] = strconv.Itoa(width)
+	msg.Args["height"] = strconv.Itoa(height)
+
+	buf, err := json.Marshal(msg)
+	if err != nil {
+		return err
+	}
+	_, err = w.Write(buf)
+
+	w.Close()
+	return err
+}
+
+func (c *consoleCmd) forwardSignal(control *websocket.Conn, sig syscall.Signal) error {
```

Comment:
Hmm, why is that a thing for console?

My understanding with console is that only SIGWINCH can be supported and that's handled by the window-resize command above.

Reply:
Yeah, I mentioned this on IRC already.

Reply:
Well, actually what's the problem here? The name of the function?

Reply:
No, just the fact that signal forwarding isn't a thing with the console at all, so none of this should exist in the first place.

The only signal that makes sense with a console is SIGWINCH and that's handled separately through the "window-resize" command above in sendTermSize.

forwardSignal() exists for forwarding signals received by the command line tool onto the running process in LXD when using exec, it simply shouldn't exist for console, nor should the "signal" command.